src/liquid/components/ld-select/test/ld-select.spec.ts

```diff
@@ -290,6 +290,48 @@ describe('ld-select', () => {
     expect(internalOptions[1].getAttribute('selected')).toEqual(null)
   })
 
+  it('emits input event on selection of an option', async () => {
+    const page = await newSpecPage({
+      components,
+      html: `
+        <ld-select placeholder="Pick a fruit" name="fruit">
+          <ld-option value="apple">Apple</ld-option>
+          <ld-option value="banana">Banana</ld-option>
+        </ld-select>
+      `,
+    })
+
+    await triggerPopperWithClick(page)
+    const { internalOptions } = await getInternalOptions(page)
+
+    const ldSelect = page.root
+
+    const spyInput = jest.fn()
+    let inputEvent
+    ldSelect.addEventListener('input', (ev) => {
+      inputEvent = ev
+      spyInput()
+    })
```

Comment:
This is a bit more complicated than it has to be. You could simply replace this with

```suggestion
    ldSelect.addEventListener('input', spyInput)
```

and later do something like

```js
expect(spyInput).toHaveBeenCalledWith(expect.objectContaining({
  detail: ["apple"],
})) // etc...
```